### DISCUSSION TITLE
how to curate grRules after getModelFromHomology

### DISCUSSION THREAD

**Discussion by FengxuSysbio** [SysBioChalmers/RAVEN#400] [Help me!] [answered]
Hi, Professor Eduard
When generating the new model based on homology with template model. How can I update and correct the grRules quickly and efficiently? Is there anything to note?
Thanks for your answers and help so much!
Kind regard,
Fengxu

**Comment by edkerk** [comment, marked as answer]
It’s not entirely clear to me what you mean with "update and correct grRules". One thing to note is that the model will contain genes starting with “OLD_”, these were subunits for which no match could be found in your organism. There is no automated way to fix this, because it inherently requires manual curation, as it might involve also looking into grRules of other reactions, compare quite similar genes etc. You can manually blast the “OLD_” gene for your organism of interest, and see if it finds a reasonable match that was perhaps just below the cutoff that you set in getModelFromHomology. It could also be that this subunit is not existing as individual gene in your organism, but is rather part of the gene coding for the other subunit of the same enzyme complex. Finally, it can also be that this whole subunit is not existing in your organism at all, as it is not essential for its functioning. As you see, it would be hard to define an automated approach to deal with this, so you should manually go through the list of “OLD_” genes.

Then, I suggest you also have a quick look at any grRule that contains multiple genes. Sometimes you end up with cases like “(gene A and gene B) or (gene A and gene B)”, which can obviously be simplified by only “gene A and gene B”. This occurs if the template models had two alternative subunits (two “versions” of gene A for instance), but your organism has only one copy (gene A).

Finally, if you know from your own experiments or literature what the grRule of specific reactions should be, you can inspect those to make sure that they’re correct.

If you’ve done gap-filling, you should also try to find genes that could catalyze those gap-filled reactions.

**Reply by FengxuSysbio** [reply]
Thanks for your suggestions.
